caravel/assets/javascripts/explorev2/components/ControlPanelsContainer.jsx

```diff
@@ -8,34 +9,38 @@ import ControlPanelSection from './ControlPanelSection';
 import FieldSetRow from './FieldSetRow';
 
 const propTypes = {
-  vizType: PropTypes.string,
-  datasourceId: PropTypes.number.isRequired,
-  datasourceType: PropTypes.string.isRequired,
+  viz_type: PropTypes.string,
+  datasource_id: PropTypes.number.isRequired,
+  datasource_type: PropTypes.string.isRequired,
   actions: PropTypes.object.isRequired,
 };
 
 const defaultProps = {
-  vizType: null,
+  viz_type: null,
 };
 
 class ControlPanelsContainer extends React.Component {
   componentWillMount() {
-    const { datasourceId, datasourceType } = this.props;
-    if (datasourceId) {
-      this.props.actions.setFormOpts(datasourceId, datasourceType);
+    const { datasource_id, datasource_type } = this.props;
+    if (datasource_id) {
+      this.props.actions.setFormOpts(datasource_id, datasource_type);
     }
   }
 
+  onChange(name, value) {
+    this.props.actions.setFormData(name, value);
```

Comment:
i might call this `setFieldValue` for clarity since it's just updating a single field.